torch/_dynamo/variables/functions.py

```diff
@@ -772,7 +772,8 @@ def call_method(
             if "grid" not in kwargs:
                 raise Unsupported("Triton kernel requires to be called with a grid")
             grid = kwargs.pop("grid")
-            return self.clone(grid=grid).call_function(tx, args, kwargs)
+            self.grid = grid
+            return self.call_function(tx, args, kwargs)
```

Comment:
I think this case mutation is actually incorrect.  You could have:
```
kernel.run(.... grid=a)
kernel.run(.... grid=b)
```
neither of these mutate `kernel`.

Reply:
I added the copy back